soccer/planning/planner/LineKickPlanner.cpp

```diff
@@ -0,0 +1,204 @@
+#include "LineKickPlanner.hpp"
+#include "planning/trajectory/RRTUtil.hpp"
+namespace Planning {
+ConfigDouble* LineKickPlanner::_approachSpeed;
+
+REGISTER_CONFIGURABLE(LineKickPlanner);
+
+void LineKickPlanner::createConfiguration(Configuration* cfg) {
+    _approachSpeed = new ConfigDouble(cfg, "LineKickPlanner/approachSpeed");
+}
+
+using namespace Geometry2d;
+Trajectory LineKickPlanner::plan(PlanRequest&& request) {
+    Point aimTarget = std::get<LineKickCommand>(request.motionCommand).target;
+    RobotInstant startInstant = request.start;
+    Ball& ball = request.context->state.ball;
+    RotationConstraints rotationConstraints = request.constraints.rot;
+    const RJ::Time curTime = RJ::now();
+    Trajectory& prevTrajectory = request.prevTrajectory;
+    if (prevTrajectory.empty()) {
+        _finalApproach = false;
+    }
+
+    // Fine segment: in the final part of the trajectory we don't replan
```

Comment:
Can you break this out into a state machine like you did with collect? I think that really cleaned things up.